src/middlewares/validateDataWithZod.ts

```diff
@@ -0,0 +1,22 @@
+import { NextFunction, Request, Response } from 'express';
+import type { LoginSchemaType } from 'src/lib/zodSchema/loginSchema';
+import type { SignUpSchemaType } from 'src/lib/zodSchema/signUpSchema';
+
+const validateDataWithZod =
+  (Schema: LoginSchemaType | SignUpSchemaType) =>
+  (req: Request & { validatedBody: Request['body'] }, res: Response, next: NextFunction) => {
+    const rawData = req.body;
+    const result = Schema.safeParse(rawData);
+
+    if (!result.success) {
+      const zodErrors = { errors: result.error.flatten().fieldErrors };
+
+      res.status(422).json(zodErrors);
+      return;
+    }
+
+    req.validatedBody = result.data;
```

Comment:
Why not just replace the body field with zod result, Typescript will force response controllers to add the validatedBody field to the request object in order to use it resulting in more code.

Reply:
Makes sense.
It was done like this owing to the fact that this schema was originally being used in js land at the time, not typescript